### ISSUE TITLE
[Intro to iOS] Twitter Redux - Ready for review!

### ISSUE THREAD

**Issue by nikrad** [nikrad/Codepath-Twitter#2]
Please review my week 4 assignment. Thanks! /cc @codepathreview

**Comment by codepathreview** [comment]
Nice app

While running it I got this exception, I think when I hit a Favorite button on my own profile page:

````
}
Optional(<Twitter.Tweet: 0x1284380a0>)
2015-10-12 12:14:15.984 Twitter[690:298513] -[Twitter.TweetCell onTapFavorite]: unrecognized selector sent to instance 0x12796bc00
2015-10-12 12:14:15.989 Twitter[690:298513] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[Twitter.TweetCell onTapFavorite]: unrecognized selector sent to instance 0x12796bc00'
*** First throw call stack:
(0x183d4cf5c 0x19923bf80 0x183d53c6c 0x183d50c14 0x183c54dcc 0x1892da3c8 0x1892da344 0x1892c2c6c 0x1892d9c5c 0x189294bdc 0x1892d3548 0x1892d2aac 0x1892a3a10 0x1892a1efc 0x183d045a4 0x183d04038 0x183d01d38 0x183c30dc0 0x18f0cc088 0x18930af44 0x10011513c 0x199a5a8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException
````

**Comment by nikrad** [comment]
Good catch! :100: I just added in the pesky selector string colon that should fix the problem.

**Comment by codepathreview** [comment]
Nice work! We added this homework so that you'll have the experience building an app that's more similar in complexity to a production app. It's also useful to have practical experience building a container view controller, which is a more advanced topic. The optional account switching feature is interesting because you have to think about how TwitterClient is caching the access token to make sure it works correctly for multiple accounts. Displaying the account switching screen is implemented using custom view controller transitions.

A few notes after checking out the code:

* You should declare all unnecessarily public properties as private for better encapsulation.
* Note that you should resolve all the Auto Layout constraint warnings in your nibs.
* Great ProfileViewController. Consider refactoring the ProfileViewController so that it shares code with the TimelineViewController. There are two general strategies: you can subclass the TimelineViewController or create a separate class called a TimelineController (not a view controller). TimelineController is a dataSource and a delegate to a UITableView and knows how to show a feed of tweets.
* Nice work encapsulating the properties of the custom cells that you added.
* Nice work abstracting the hamburger controller so that it can be reused in other applications.
* Great job with the blurred header. There are several approaches here that are all valid. You can implement the header has a table header, a section header, or a cell. My preferred approach is to implement it as a table header, it's the easiest way to flow the image above the content. These detail touches are where you really learn the nooks and crannies of the framework.


Congratulations on finishing the final assignment! It's impressive to realize that in just four weeks, you've gone from building a simple tip calculator app to a polished MVC client.

-- dirk
